test/e2e/tests/test_monitoring_schedule.py

```diff
@@ -0,0 +1,219 @@
+            wait_sagemaker_monitoring_schedule_status(
+                sagemaker_client, schedule_name, expected_status
+            )
+            == wait_resource_monitoring_schedule_status(reference, expected_status, 2)
+            == expected_status
+        )
+
+    def test_smoke(self, sagemaker_client, xgboost_churn_data_quality_monitoring_schedule):
+        (reference, resource, spec) = xgboost_churn_data_quality_monitoring_schedule
+        assert k8s.get_resource_exists(reference)
+
+        monitoring_schedule_name = resource["spec"].get("monitoringScheduleName")
+
+        assert (
+            k8s.get_resource_arn(resource)
+            == describe_sagemaker_monitoring_schedule(
+                sagemaker_client, monitoring_schedule_name
+            )["MonitoringScheduleArn"]
+        )
+
+        # scheule transitions Pending -> Scheduled state
+        # Pending status is shortlived only for 30 seconds because baselining job has already been run
+        # remove the checks for Pending status if the test is flaky because of this
+        # as the main objective is to test for Scheduled status
+        # OR
+        # create the schedule with a on-going baseline job where it waits for the baselining job to complete
+        assert (
+            wait_resource_monitoring_schedule_status(
+                reference, self.status_pending, 5, 2
+            )
+            == self.status_pending
+        )
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "False", 5, 2)
+
+        self._assert_monitoring_schedule_status_in_sync(
+            sagemaker_client, monitoring_schedule_name, reference, self.status_scheduled
+        )
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True")
+
+        # Update the resource
+        new_cron_expression = "cron(0 * * * ? *)"
+        spec["spec"]["monitoringScheduleConfig"]["scheduleConfig"][
+            "scheduleExpression"
+        ] = new_cron_expression
+        resource = k8s.patch_custom_resource(reference, spec)
+        resource = k8s.wait_resource_consumed_by_controller(reference)
+        assert resource is not None
+
+        self._assert_monitoring_schedule_status_in_sync(
+            sagemaker_client, monitoring_schedule_name, reference, self.status_scheduled
+        )
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True")
+
+        latest_schedule = describe_sagemaker_monitoring_schedule(
+            sagemaker_client, monitoring_schedule_name
+        )
+        assert (
+            latest_schedule["MonitoringScheduleConfig"]["ScheduleConfig"][
+                "ScheduleExpression"
+            ]
+            == new_cron_expression
+        )
+
+        # Delete the k8s resource.
+        _, deleted = k8s.delete_custom_resource(reference)
+        assert deleted
+
+        # 30 sec wait for server-side acknowledgement
+        schedule_deleted = False
+        for _ in range(3):
+            time.sleep(10)
```

Comment:
30 seconds for acknowledgement? That's a long time between API call and response

Reply:
edited the comment
its wait for server-side clean up